Xamarin.Forms.Core/View.cs

```diff
@@ -155,9 +155,25 @@ public LayoutOptions VerticalOptions
 		protected override void OnBindingContextChanged()
 		{
 			this.PropagateBindingContext(GestureRecognizers);
+			PropagateBindingContextToStateTriggers();
```

Comment:
Should this be at the VisualElement level?

*VisualStateManager.VisualStateGroupsProperty*

Can be applied to anything that's of type VisualElement

```C#
public static readonly BindableProperty VisualStateGroupsProperty =
			BindableProperty.CreateAttached("VisualStateGroups", typeof(VisualStateGroupList), typeof(VisualElement),
				defaultValue: null, propertyChanged: VisualStateGroupsPropertyChanged,
				defaultValueCreator: bindable => new VisualStateGroupList { VisualElement = (VisualElement)bindable });
```

Reply:
Yeah, better. I moved it.